_posts/en/newsletters/2025-10-24-newsletter.md

```diff
@@ -0,0 +1,151 @@
+---
+title: 'Bitcoin Optech Newsletter #377'
+permalink: /en/newsletters/2025/10/24/
+name: 2025-10-24-newsletter
+slug: 2025-10-24-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+FIXME:bitschmidty
+
+## News
+
+- **Detecting block template feerate increases using cluster mempool:** Abubakar Sadiq Ismail recently [posted][ismail post] to Delving Bitcoin about the possibility to track the potential fee increase from each mempool update in order to provide a new block template to miners only when the fee rate improvement justifies it. This approach would reduce the number of redundant block template builds, which can delay transaction processing and relay to peers. The proposal leverages [cluster mempool][topic cluster mempool] to assess whether a potential fee rate improvement is available without a full rebuild of a block template.
+
+  When a new transaction enters the mempool, it connects to zero or more existing clusters triggering re-linearization (see Newsletter [#312][news312 lin] for more information about linearization) of affected clusters to produce old (pre-update) and new (post-update) feerate diagrams. The old diagram identifies potential chunks for eviction from the block template, while the new diagram identifies high-feerate chunks for potential addition. The system then follows a 4-step process to simulate the impact:
+  1. Eviction: Remove matching chunks from a template copy, updating modified fees and sizes.
+
+  2. Naive Merge: Greedily add candidate chunks while respecting block weight limits to estimate potential fee gains (ΔF).
+
+  3. Iterative Merge: If the naive estimate is inconclusive, use a more detailed simulation to refine ΔF.
+
+  4. Decision: Compare ΔF to a threshold; if it exceeds the threshold, rebuild the full block template and send it to miners. Otherwise, skip to avoid unnecessary computation.
```

Comment:
```suggestion
  4. Decision: Compare ΔF to a threshold; if it exceeds the threshold, rebuild the block template and send it to miners. Otherwise, skip to avoid unnecessary computation.
```